pkg/server/server.go

```diff
@@ -384,6 +393,9 @@ func (h *handler) doMapping(identity *token.Identity) (string, []string, error)
 			if err != nil {
 				return "", nil, fmt.Errorf("mapper %s renderTemplates error: %v", m.Name(), err)
 			}
+			if m.UsernamePrefixEnforce() && ReservedPrefixExists(username, m.UsernamePrefixReserveList()) {
```

Comment:
Can you add a unit test for this too?

Reply:
have added it https://github.com/kubernetes-sigs/aws-iam-authenticator/pull/568/files#diff-fe7a244e18e92d8f347af17363209172e1d79224f086b0585a1bbeef3b626c4eR172